### PR TITLE
base_fw: update mem_state_info TLV

### PR DESCRIPTION
**Pull request by dnikodem**
This change improves the TLV assembly algorithm in accordance with how the part for hpsram is assembled. In the case when ebb_state_dword_count is equal to 0, we skip adding the LSPGCTL register value for lpsram.

The following PR must be merged into the Zephyr repository before we can merge this PR:
- [x] https://github.com/zephyrproject-rtos/zephyr/pull/70221

### CODE REVIEW

File: src/platform/lunarlake/include/platform/lib/shim.h
Comment (lgirdwood) [inline src/platform/lunarlake/include/platform/lib/shim.h:50]: @dnikodem can we read this from Zephyr headers today ?

Comment (dnikodem) [reply src/platform/lunarlake/include/platform/lib/shim.h:50]: ok, I will update this PR, thanks.

**Comment by dnikodem** [comment]
Added https://github.com/zephyrproject-rtos/zephyr/pull/70221 - this is the first step to clean up SOF headers.

File: src/audio/base_fw.c
Comment (lyakh) [inline src/audio/base_fw.c:233]: this of course changes the behaviour significantly. Before this commit `index` was incremented twice in lines 226 and 228, now it isn't any more. Is this intended? Was that a bug?

Comment (dnikodem) [reply src/audio/base_fw.c:233]: The tuple_data should contain the correct number of LPSRAM PGCTL register values (in the same way as it is done for HPSRAM). In the previous solution, if the number of ebb_state_dword_count equaled 0, a dword was added in tuple_data with the PGCTL status value for a non-existing LPSRAM bank. 

Therefore, the parser that is using data from tuple_data should not expect an "extra" LSPGCTL status value in the tuple_data, and then it could misinterpret it as some other type of data or status.

Comment (lyakh) [reply src/audio/base_fw.c:234]: @dnikodem right, but if `ebb_state_dword_count != 0`? Then you'd add several register values into `tuple_data` without incrementing `index`, so after the loop you'd overwrite the first of those values... I think you actually wanted to do
```
	for (i = 0; i < info.ebb_state_dword_count; i++) {
#ifdef INTEL_ADSP
		tuple_data[index++] = LPSRAM_REGS(i)->USxPGCTL;
#else
		tuple_data[index++] = 0;
#endif
	}
	if (info.ebb_state_dword_count)
		tuple_data[index++] = info.page_alloc_struct.page_alloc_count;
```

**Review by lyakh — CHANGES_REQUESTED** [review]

**Review by lgirdwood — COMMENTED** [review]
@dnikodem I assume you will make non draft after the Zephyr part is merged and we can rebase/merge.

**Comment by dnikodem** [comment]
> @dnikodem I assume you will make non draft after the Zephyr part is merged and we can rebase/merge.

correct, we need firstly zephyr part merged.

File: src/audio/base_fw.c
Comment (dnikodem) [reply src/audio/base_fw.c:233]: @lyakh Fixed, thanks

**Comment by kv2019i** [comment]
Oops, @dnikodem  I didn't realize this PR was still open as well. I was intending to get he pending base_fw changes out of the wait, but missed this one. This will now conflict with https://github.com/thesofproject/sof/pull/9103 

Let's see which gets the reviews together first, and then rebase the other one.

**Comment by lgirdwood** [comment]
> Oops, @dnikodem I didn't realize this PR was still open as well. I was intending to get he pending base_fw changes out of the wait, but missed this one. This will now conflict with #9103
> 
> Let's see which gets the reviews together first, and then rebase the other one.

oh, yeah - this one was blocked on Zephyr PR merging (now merged). @dnikodem do we need a west update ? I think @kv2019i did a west update a few days ago so we may be good just to rebase this ?

File: src/audio/base_fw_intel.c
Comment (lgirdwood) [inline src/audio/base_fw_intel.c:134]: @kv2019i do we still need the ifdef now ? since this file is Intel only ?

**Review by lgirdwood — COMMENTED** [review]
Just one question on ifdefs, but otherwise LGTM.

File: src/audio/base_fw_intel.c
Comment (kv2019i) [reply src/audio/base_fw_intel.c:134]: Ack @lgirdwood , this can be dropped now.

File: src/audio/base_fw_intel.c
Comment (kv2019i) [inline src/audio/base_fw_intel.c:160]: Likewise here.

**Review by kv2019i — COMMENTED** [review]
Thanks @dnikodem , this is a welcome update. Code looks good, but I think you can go a bit further and drop the ifdefs as this code is now only compiled for Intel targets that have enabled the IPC4 basefw Intel extensions (so should cover all platforms now covered by INTEL_ADSP ifdef guard).

**Review by kv2019i — APPROVED** [review]
Thanks, looks good now!

File: src/audio/base_fw_intel.c
Comment (dnikodem) [reply src/audio/base_fw_intel.c]: Fixed, thanks :)

**Review by lgirdwood — APPROVED** [review]
Nice to remove all the old macros :)

**Pull request merged**
